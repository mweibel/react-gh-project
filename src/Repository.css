.button.button-link {
  padding: 0;
  margin: 0;
  line-height: normal;
  height: auto;
}