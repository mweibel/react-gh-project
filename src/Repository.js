import React from 'react'
import PropTypes from 'prop-types'

const Repository = ({ name, index, onClick, active = false, loading = false, issues = []}) => (
  <li onClick={(event) => onClick(event, index, name)}>
    {name}
    {active && (
      <ul>
        {loading && <li>Loading</li>}
        {issues.map(({ id, title, html_url }) => <li key={id}><a href={html_url}>{title}</a></li>)}
      </ul>
    )}
  </li>
)
Repository.propTypes = {
  name: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool,
  loading: PropTypes.bool,
  issues: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired
  }).isRequired)
}


export default Repository