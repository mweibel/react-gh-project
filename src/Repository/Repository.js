import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Loading } from '../Loading/Loading';
import './Repository.css';
import { Issue } from '../Issue/Issues';

const Repository = ({
  name,
  index,
  onClick,
  active = false,
  loading = false,
  issues = []
}) => (
  <Fragment>
    <button
      type="button"
      className="button button-clear button-link"
      onClick={event => onClick(event, index, name)}
    >
      {name}
    </button>
    {active && (
      <ul>
        {loading ? (
          <li>
            <Loading />
          </li>
        ) : (
          issues.map(issue => (
            <li key={issue.id}>
              <Issue {...issue} />
            </li>
          ))
        )}
      </ul>
    )}
  </Fragment>
);
Repository.propTypes = {
  name: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool,
  loading: PropTypes.bool,
  issues: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      html_url: PropTypes.string.isRequired
    }).isRequired
  )
};

export default Repository;
