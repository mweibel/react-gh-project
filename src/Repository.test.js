import React from 'react';
import ReactDOM from 'react-dom';
import Repository from './Repository';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const props = {
    name: 'test-repo',
    index: 0,
    onClick: jest.fn()
  };

  const tree = renderer.create(
    <Repository {...props} />
  );
  expect(tree).toMatchSnapshot();
});
