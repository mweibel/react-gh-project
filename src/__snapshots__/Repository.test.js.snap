// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render the issues if active is false 1`] = `
<li
  onClick={[Function]}
>
  test-repo
</li>
`;

exports[`renders while loading 1`] = `
<li
  onClick={[Function]}
>
  test-repo
  <ul>
    <li>
      Loading
    </li>
  </ul>
</li>
`;

exports[`renders with a list of issues 1`] = `
<li
  onClick={[Function]}
>
  test-repo
  <ul>
    <li>
      <a
        href="https://example.org/testissue1"
      >
        testissue1
      </a>
    </li>
    <li>
      <a
        href="https://example.org/testissue2"
      >
        testissue2
      </a>
    </li>
  </ul>
</li>
`;

exports[`renders with minimal props 1`] = `
<li
  onClick={[Function]}
>
  test-repo
</li>
`;
